lib/pbench/server/database/models/datasets.py

```diff
@@ -666,7 +667,10 @@ class Metadata(Database.Base):
         value       Metadata value string
     """
 
-    __tablename__ = "dataset_metadata"
+    # Typically the table name is the plural form of the class name, but here
+    # 'metadatas', while plural, could apply to anything. Adding the prefix of
+    # 'dataset' communicates the nature of the table data.
+    __tablename__ = "dataset_metadatas"
```

Comment:
I second this emotion:  "metadatas" is even more of a strain than "metadatum"...let's just go with "metadata".

Reply:
I'm confused...what is wrong with `"dataset_metadata"`?  (I.e., the way the code _was_ before this change....)

Reply:
I'd don't know.  Why are we reviewing a Draft PR in such detail?

Reply:
Because

> In draft mode until PR https://github.com/distributed-system-analysis/pbench/pull/3174 lands.

and #3174 has landed!  😆

Reply:
While this is still in `Draft` mode as I have a few updates to make, I believe this concern has been resolved.

Reply:
Indeed.

Conversationally, I'll just comment here that my preference for plural tables has waned slightly through the many tedious hours I've recently spent staring at SQLAlchemy-generated queries in my efforts to debug my bizarre problems with `sqlite3` joined queries. The problem is that, presented with a join, SQLAlchemy nicely decorates column names with table names, to avoid ambiguity:

```
QUERY SELECT DISTINCT datasets.access, datasets.id, datasets.name, datasets.owner_id, datasets.resource_id, datasets.uploaded 
FROM datasets JOIN dataset_metadata ON datasets.id = dataset_metadata.dataset_ref 
WHERE datasets.owner_id = '20' OR datasets.access = 'public' ORDER BY datasets.resource_id
 LIMIT 5
```

The "FROM datasets" is the part where the plural has always felt "right" to me, as we're searching a collection of "datasets" (plural) to find individual *instances* of "dataset" (singular). And without the join, the `SELECT` and `WHERE` clauses can use the raw column names like `access`, `id`. The catch here is that, with the join, we columns from both the `datasets` and `dataset_metadata` tables, which needs to be qualified in SQL.

So even though the actual `WHERE` and `SELECT` expressions are against each specific *instance* of "dataset" (singular) they're now fully qualified and I find `WHERE datasets.owner_id = '20'` odd and tedious.

Ultimately, I reluctantly conclude, one can't win this game and I'm gradually moving towards losing my preference entirely.

By which I'm not suggesting any action for this PR... just "making conversation". 🙄